src/Symfony/Component/Messenger/Bridge/Amqp/Transport/Connection.php

```diff
@@ -418,18 +423,18 @@ public function get(string $queueName): ?\AMQPEnvelope
     {
         $this->clearWhenDisconnected();
 
-        if ($this->shouldSetup()) {
-            $this->setupExchangeAndQueues();
+        if ($this->autoSetup) {
+            $this->setup();
         }
 
         try {
             if (false !== $message = $this->queue($queueName)->get()) {
                 return $message;
             }
         } catch (\AMQPQueueException $e) {
-            if (404 === $e->getCode() && $this->shouldSetup()) {
+            if (404 === $e->getCode() && $this->autoSetup) {
                 // If we get a 404 for the queue, it means we need to set up the exchange & queue.
-                $this->setupExchangeAndQueues();
+                $this->setup();
```

Comment:
This call looks like unreacheable. 
By this if `autoSetup` was `true` - setup was called at line 427 and `autoSetup` now became `false`.

Reply:
Mybe will be better ro remove this catch ?
I suspect that it lead to infinity loop.

Reply:
> This call looks like unreacheable

hmm.. you're right. This should hav been setup at the beggining of the method